project-management/src/main/java/life/qbic/projectmanagement/application/ProjectInformationService.java

```diff
@@ -204,7 +207,54 @@ public void removeFunding(ProjectId projectId) {
     projectRepository.update(project);
   }
 
-  public void updateModifiedDate(ProjectId projectID, Instant modifiedOn) {
-    projectRepository.unsafeUpdateLastModified(projectID, modifiedOn);
+  public void updateModifiedDate(ProjectId projectID, Instant modifiedOn) throws ProjectNotFoundException {
+    // The update might fail due to an optimistic locking exception (concurrent access of other processes)
+    // To address this, the update is tried until it will eventually not throw the locking exception anymore.
+    // This approach naively assumes, that the locked resource will be released eventually again by the other process.
+    var attempt = 1;
+    var project = projectRepository.find(projectID).orElseThrow(() -> new ProjectNotFoundException("Project with not found: %s".formatted(projectID)));
+    while(true) {
+      try {
+        tryToUpdateModifiedDate(project, modifiedOn);
+        return;
+      } catch (OptimisticEntityLockException e) {
+        log.debug("Optimistic lock exception occurred while updating modified date for project " + projectID);
+      }
+      try {
+        Thread.sleep(calcBase2Duration(attempt));
+      } catch (InterruptedException e) {
+        log.error("Interrupted while updating modified date for project " + projectID);
+        // We try one last time
+        tryToUpdateModifiedDate(project, modifiedOn);
```

Comment:
Why do we try one last time? What happens if an OptimisticEntityLockException is thrown here? The thread interrruption is then ignored?

Reply:
No, if the Thread is interrupted and the task has not yet completed we must try again. Otherwise it will get lost.